estate/models/estate_property.py

```diff
@@ -0,0 +1,137 @@
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection([
+        ('north', 'North'),
+        ('south', 'South'),
+        ('east', 'East'),
+        ('west', 'West'),
+    ])
+    active = fields.Boolean(default=True)
+    state = fields.Selection([
+        ('new', 'New'),
+        ('offer_received', 'Offer Received'),
+        ('offer_accepted', 'Offer Accepted'),
+        ('sold', 'Sold'),
+        ('cancelled', 'Cancelled'),
+    ], default='new', required=True, copy=False)
+    property_type_id = fields.Many2one(
+        "estate.property.type", string="Property Type")
+    buyer_id = fields.Many2one("res.partner", string="Buyer",
+                               copy=False, compute="_compute_buyer")
+    salesperson_id = fields.Many2one(
+        "res.users", string="Salesperson", default=lambda self: self.env.uid)
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many(
+        "estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Integer(
+        compute="_compute_total_area", string="Total Area (sqm)")
+    best_offer = fields.Float(
+        compute="_compute_best_offer")
+
+    _check_expected_price = models.Constraint(
+        'CHECK(expected_price > 0)', 'The expected price must be strictly positive.'
+    )
+    _check_selling_price = models.Constraint(
+        'CHECK(selling_price >= 0)', 'The selling price must be non-negative.'
+    )
+
+    @api.depends('offer_ids.status', 'offer_ids.price')
+    def _compute_selling_price(self):
+        for property in self:
+            accepted_offer = property.offer_ids.filtered(
+                lambda o: o.status == 'accepted')
+            if accepted_offer:
+                property.selling_price = accepted_offer.price
+            else:
+                property.selling_price = 0.0
+
+    @api.depends('offer_ids.status', 'offer_ids.partner_id')
+    def _compute_buyer(self):
+        for property in self:
+            accepted_offer = property.offer_ids.filtered(
+                lambda o: o.status == 'accepted')
+            if accepted_offer:
+                property.buyer_id = accepted_offer.partner_id
+            else:
+                property.buyer_id = None
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for property in self:
+            property.total_area = property.living_area + property.garden_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for property in self:
+            if property.offer_ids:
+                property.best_offer = max(property.offer_ids.mapped('price'))
+            else:
+                property.best_offer = 0.0
```

Comment:
```suggestion
            property.best_offer = max(property.offer_ids.mapped('price')) if property.offer_ids else 0.0
```
This is good but you could make it shorter. A bit of sugar, not required :)